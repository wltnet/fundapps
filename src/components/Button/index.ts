import Button from "./Button";

export interface Props {
  title: string;
  onClick: () => void;
  className?: string;
}

export default Button;
