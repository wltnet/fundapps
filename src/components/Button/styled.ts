import styled from "styled-components";

export const Wrapper = styled.button`
  border-radius: 5px;
  padding: 15px;
  font-size: 1rem;
  font-weight: 700;
  border: 2px solid #ff1493;
  color: #ff1493;
  background: #fff;
  cursor: pointer;
`;
