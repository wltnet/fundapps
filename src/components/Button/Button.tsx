import * as React from "react";
import { Wrapper } from "./styled";
import { Props } from ".";

const Button = ({ title, onClick, className }: Props): React.ReactElement => (
  <Wrapper className={className} onClick={onClick}>
    {title}
  </Wrapper>
);

export default Button;
