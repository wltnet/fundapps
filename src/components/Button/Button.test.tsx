import React from "react";
import { shallow } from "enzyme";
import Button from "./Button";

jest.mock("./styled", () => ({ Wrapper: "Wrapper" }));

describe("<Button />", () => {
  it("should match the snapshot", () => {
    const props = {
      title: "Show more",
      onClick: jest.fn(),
    };
    const wrapper = shallow(<Button {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
