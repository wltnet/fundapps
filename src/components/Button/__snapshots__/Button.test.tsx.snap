// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should match the snapshot 1`] = `
<Wrapper
  onClick={[MockFunction]}
>
  Show more
</Wrapper>
`;
